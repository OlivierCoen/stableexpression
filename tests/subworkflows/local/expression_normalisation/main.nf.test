nextflow_workflow {

    name "Test Workflow EXPRESSION_NORMALISATION"
    script "subworkflows/local/expression_normalisation/main.nf"
    workflow "EXPRESSION_NORMALISATION"
    tag "subworkflow_expression_normalisation"
    tag "subworkflow"

    test("DESeq2 Normalisation") {

        when {
            workflow {
                """
                rnaseq_raw_file = file( '$projectDir/tests/test_data/custom_datasets/rnaseq.raw.csv', checkIfExists: true )
                rnaseq_raw_design_file = file( '$projectDir/tests/test_data/custom_datasets/rnaseq.raw.design.csv', checkIfExists: true )
                microarray_normalised_file = file( '$projectDir/tests/test_data/custom_datasets/microarray.normalised.csv', checkIfExists: true )
                microarray_normalised_design_file = file( '$projectDir/tests/test_data/custom_datasets/microarray.normalised.design.csv', checkIfExists: true )
                ch_datasets = Channel.of(
                    [ [normalised: false, design: rnaseq_raw_design_file, dataset: "rnaseq_raw", platform: "rnaseq"], rnaseq_raw_file],
                    [ [normalised: true, design: microarray_normalised_design_file, dataset: "microarray_normalised", platform: "microarray"], microarray_normalised_file ]
                )
                normalisation_method = "deseq2"
                input[0] = ch_datasets
                input[1] = normalisation_method
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("EdgeR Normalisation") {

        when {
            workflow {
                """
                rnaseq_raw_file = file( '$projectDir/tests/test_data/custom_datasets/rnaseq.raw.csv', checkIfExists: true )
                rnaseq_raw_design_file = file( '$projectDir/tests/test_data/custom_datasets/rnaseq.raw.design.csv', checkIfExists: true )
                microarray_normalised_file = file( '$projectDir/tests/test_data/custom_datasets/microarray.normalised.csv', checkIfExists: true )
                microarray_normalised_design_file = file( '$projectDir/tests/test_data/custom_datasets/microarray.normalised.design.csv', checkIfExists: true )
                ch_datasets = Channel.of(
                    [ [normalised: false, design: rnaseq_raw_design_file, dataset: "rnaseq_raw", platform: "rnaseq"], rnaseq_raw_file],
                    [ [normalised: true, design: microarray_normalised_design_file, dataset: "microarray_normalised", platform: "microarray"], microarray_normalised_file ]
                )
                normalisation_method = "edger"
                input[0] = ch_datasets
                input[1] = normalisation_method
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

    test("No rnaseq normalisation") {

        when {
            workflow {
                """
                microarray_normalised_file = file( '$projectDir/tests/test_data/custom_datasets/microarray.normalised.csv', checkIfExists: true )
                microarray_normalised_design_file = file( '$projectDir/tests/test_data/custom_datasets/microarray.normalised.design.csv', checkIfExists: true )
                ch_datasets = Channel.of(
                    [ [normalised: true, design: microarray_normalised_design_file, dataset: "microarray_normalised", platform: "microarray"], microarray_normalised_file ]
                )
                normalisation_method = "deseq2"
                input[0] = ch_datasets
                input[1] = normalisation_method
                """
            }
        }

        then {
            assert workflow.success
            assert snapshot(workflow.out).match()
        }

    }

}
