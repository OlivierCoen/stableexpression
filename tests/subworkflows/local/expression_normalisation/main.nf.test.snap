{
    "No rnaseq normalisation": {
        "content": [
            {
                "0": [
                    [
                        {
                            "normalised": true,
                            "design": "microarray.normalised.design.csv:md5,9662cc2f58d86cc552d6ff9cf094dd67",
                            "dataset": "microarray_normalised",
                            "platform": "microarray"
                        },
                        "microarray.normalised.quant_norm.parquet:md5,eabdc05374b0c21ebcfd83b01efcedd4"
                    ]
                ],
                "normalised_counts": [
                    [
                        {
                            "normalised": true,
                            "design": "microarray.normalised.design.csv:md5,9662cc2f58d86cc552d6ff9cf094dd67",
                            "dataset": "microarray_normalised",
                            "platform": "microarray"
                        },
                        "microarray.normalised.quant_norm.parquet:md5,eabdc05374b0c21ebcfd83b01efcedd4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-05T15:25:45.682092837"
    },
    "EdgeR Normalisation": {
        "content": [
            {
                "0": [
                    [
                        {
                            "normalised": false,
                            "design": "rnaseq.raw.design.csv:md5,39470b02a211aff791f9e4851b017488",
                            "dataset": "rnaseq_raw",
                            "platform": "rnaseq"
                        },
                        "rnaseq.raw.cpm.quant_norm.parquet:md5,5c9765f2ffbc78fb90b5591a28e68fe4"
                    ],
                    [
                        {
                            "normalised": true,
                            "design": "microarray.normalised.design.csv:md5,9662cc2f58d86cc552d6ff9cf094dd67",
                            "dataset": "microarray_normalised",
                            "platform": "microarray"
                        },
                        "microarray.normalised.quant_norm.parquet:md5,eabdc05374b0c21ebcfd83b01efcedd4"
                    ]
                ],
                "normalised_counts": [
                    [
                        {
                            "normalised": false,
                            "design": "rnaseq.raw.design.csv:md5,39470b02a211aff791f9e4851b017488",
                            "dataset": "rnaseq_raw",
                            "platform": "rnaseq"
                        },
                        "rnaseq.raw.cpm.quant_norm.parquet:md5,5c9765f2ffbc78fb90b5591a28e68fe4"
                    ],
                    [
                        {
                            "normalised": true,
                            "design": "microarray.normalised.design.csv:md5,9662cc2f58d86cc552d6ff9cf094dd67",
                            "dataset": "microarray_normalised",
                            "platform": "microarray"
                        },
                        "microarray.normalised.quant_norm.parquet:md5,eabdc05374b0c21ebcfd83b01efcedd4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-05T15:26:20.327117737"
    },
    "DESeq2 Normalisation": {
        "content": [
            {
                "0": [
                    [
                        {
                            "normalised": false,
                            "design": "rnaseq.raw.design.csv:md5,39470b02a211aff791f9e4851b017488",
                            "dataset": "rnaseq_raw",
                            "platform": "rnaseq"
                        },
                        "rnaseq.raw.cpm.quant_norm.parquet:md5,5c9765f2ffbc78fb90b5591a28e68fe4"
                    ],
                    [
                        {
                            "normalised": true,
                            "design": "microarray.normalised.design.csv:md5,9662cc2f58d86cc552d6ff9cf094dd67",
                            "dataset": "microarray_normalised",
                            "platform": "microarray"
                        },
                        "microarray.normalised.quant_norm.parquet:md5,eabdc05374b0c21ebcfd83b01efcedd4"
                    ]
                ],
                "normalised_counts": [
                    [
                        {
                            "normalised": false,
                            "design": "rnaseq.raw.design.csv:md5,39470b02a211aff791f9e4851b017488",
                            "dataset": "rnaseq_raw",
                            "platform": "rnaseq"
                        },
                        "rnaseq.raw.cpm.quant_norm.parquet:md5,5c9765f2ffbc78fb90b5591a28e68fe4"
                    ],
                    [
                        {
                            "normalised": true,
                            "design": "microarray.normalised.design.csv:md5,9662cc2f58d86cc552d6ff9cf094dd67",
                            "dataset": "microarray_normalised",
                            "platform": "microarray"
                        },
                        "microarray.normalised.quant_norm.parquet:md5,eabdc05374b0c21ebcfd83b01efcedd4"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-02-05T15:25:35.777940484"
    }
}